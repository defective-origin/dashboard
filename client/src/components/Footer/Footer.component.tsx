import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Block, { BlockProps } from 'components/Block'

// ---| self |---
import css from './Footer.module.scss'

export type FooterProps = BlockProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <Footer />
 */
export function Footer(props: FooterProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.Footer, className)

  return <Block className={_className} area='bottom' v='x' {...otherProps}>{children}</Block>
}

Footer.displayName = 'Footer'

export default Footer
