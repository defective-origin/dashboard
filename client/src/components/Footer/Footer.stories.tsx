// eslint-disable-next-line no-restricted-imports
import { field, params } from '../../../.storybook/tool'
import type { Meta, StoryObj } from '@storybook/react'
import Footer from './Footer.component'

const meta: Meta<typeof Footer> = {
  component: Footer,
  title: 'Components/LAYOUT/MAGNETS/Footer',
  tags: ['autodocs'],
  argTypes: {
    className: field.string(),
    children: field.reactNode(),
  },
}

export default meta

type Story = StoryObj<typeof Footer>

export const Demo: Story = {
  parameters: params('Footer'),
  args: {
    className: '',
    children: 'Demo',
  },
}
