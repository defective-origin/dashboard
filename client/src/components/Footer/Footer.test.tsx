// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Footer from './Footer.component'
// import useFooter from './Footer.hook'
// import FooterProvider, Footer from './Footer.context'

describe('[Footer] component', () => {
  it('should render component', () => {
    const container = render(<Footer />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})

// describe('[Footer] hook', () => {
//   it('should return options', () => {
//     const { result } = renderHook(() => useFooter())
// 
//     expect(result.current).toEqual(null)
//   })
// })

// describe('[Footer] context', () => {
//   const wrapper = (props: FooterProps) => <FooterProvider { ...props } />
// 
//   it('should return options', () => {
//     const { result } = renderHook(() => useFooter(), { wrapper })
// 
//     expect(result.current).toBeTruthy()
//   })
// })
