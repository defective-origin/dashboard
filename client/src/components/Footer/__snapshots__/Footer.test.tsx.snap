// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Footer] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="block block--x item _Footer_33bf66"
    style="grid-area: bottom;"
  />
</DocumentFragment>
`;
